app/controllers/application_controller.rb

```diff
@@ -7,13 +7,24 @@ class ApplicationController < ActionController::Base
 
   before_action :is_logged_in
   before_action :setup_query
-  before_action :refresh_urs_if_needed, except: [:logout, :refresh_token]
+  before_action :refresh_urs_if_needed, except: [:logout, :refresh_token] # URS login
   before_action :provider_set?
 
   rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
 
   protected
 
+  # Seconds ahead of the token expiration that the server should
+  # attempt to refresh the token
+  SERVER_EXPIRATION_OFFSET_S = 60
+  # For testing, token expires after 10 seconds
+  # SERVER_EXPIRATION_OFFSET_S = 3590
+
+  def urs_login_required?
+    ENV['urs_login_required'] == 'true'
```

Comment:
I think it might make sense for this to default to true. That will save work for the SA team for having to change environment variables until we actually need to change this.